src/test/java/com/aws/greengrass/clientdevices/auth/certificate/BackgroundCertificateRefreshTest.java

```diff
@@ -94,16 +113,30 @@ void setup() throws DeviceConfigurationException, KeyStoreException {
         configurationTopics.getContext().put(VerifyIotCertificate.class, verifyIotCertificateMock);
         useCases.init(configurationTopics.getContext());
 
-        backgroundRefresh = new BackgroundCertificateRefresh(
+        this.clockMock = Optional.empty();
+
+        backgroundRefresh = spy(new BackgroundCertificateRefresh(
                 schedulerMock, thingRegistry, networkStateMock, 
-                certificateRegistry, pemStore, iotAuthClientFake, useCases);
+                certificateRegistry, pemStore, iotAuthClientFake, useCases));
     }
 
     @AfterEach
     void cleanup() throws IOException {
+        this.clockMock.ifPresent(ScopedMock::close);
         configurationTopics.getContext().close();
     }
 
+    @SuppressWarnings("PMD.CloseResource")
+    private void mockInstant(long expected) {
```

Comment:
thought for a potential follow-up:   as an alternative, we could have our own implementation of `Clock` that lets us mutate the instant and then inject that in `BackgroundCertificateRefresh`.   Would express intent more clearly and avoid the need to override standard library methods :)

Reply:
https://github.com/mockito/mockito/wiki/Using-Spies-(and-Fakes)#case-study-2-to-test-a-jobscheduler-that-uses-scheduledexecutorservice has an example